File: 104 Maximum Depth of Binary Tree/C/solution.c
#include <stdlib.h>

#include "solution.h"

int max(int a, int b);

int maxDepth(struct TreeNode* root) {
    if (root == NULL) {
        return 0;
    }

    int left = maxDepth(root->left);
    int right = maxDepth(root->right);
    return max(left, right) + 1;
}

int max(int a, int b) {
    return a >= b ? a : b;
}

File: 104 Maximum Depth of Binary Tree/Python/Solution.py
from TreeNode import TreeNode

from typing import Optional


class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root == None:
            return 0
        
        left: int = self.maxDepth(root.left)
        right: int = self.maxDepth(root.right)
        return 1 + max(left, right)
            